http-server/src/tests.rs

```diff
@@ -308,3 +314,23 @@ async fn can_register_modules() {
 	assert_eq!(err.to_string(), expected_err.to_string());
 	assert_eq!(server.method_names().len(), 2);
 }
+
+#[tokio::test]
+async fn stop_works() {
+	let _ = env_logger::try_init();
+	let (_addr, join_handle, mut stop_handle) = server_with_handles().with_default_timeout().await.unwrap();
+	stop_handle.stop().with_default_timeout().await.unwrap().unwrap();
+	stop_handle.wait_for_stop().with_default_timeout().await.unwrap();
+
+	// After that we should be able to wait for task handle to finish.
+	// First `unwrap` is timeout, second is `JoinHandle`'s one, third is the server future result.
+	join_handle
+		.with_default_timeout()
+		.await
+		.expect("Timeout")
+		.expect("Join error")
+		.expect("Server stopped with an error");
+
+	// After server was stopped, attempt to stop it again should result in an error.
+	assert!(stop_handle.stop().with_default_timeout().await.unwrap().is_err());
```

Comment:
Can we assert that we actually get the `AlreadyStopped` error here? I get that it is the only possible error, but if the code is ever changed in the future this may not be true anymore.